src/charm.py

```diff
@@ -702,17 +753,16 @@ def mongo_config(self) -> MongoConfiguration:
     @property
     def mongos_config(self) -> MongoConfiguration:
         """Generates a MongoDBConfiguration object for mongos in the deployment of MongoDB."""
-        hosts = self.get_mongos_hosts()
         external_ca, _ = self.tls.get_tls_files(internal=False)
         internal_ca, _ = self.tls.get_tls_files(internal=True)
 
         return MongoConfiguration(
             database=self.database,
             username=self.get_secret(APP_SCOPE, Config.Secrets.USERNAME),
             password=self.get_secret(APP_SCOPE, Config.Secrets.PASSWORD),
-            hosts=hosts,
+            hosts=self.get_mongos_hosts(),
```

Comment:
could use second pair of eyes on this bit. 

It has big implications because now mongodb charm will operate (for admin operations) with the external IP.

I don't believe it's a big issue, since latency for admin operations isn't a priority(latency for clients is). But maybe I am missing something

Reply:
After some thought I do not believe we should do this.

I believe charm should use its own internal hosts

Reply:
Yes it should, we should not operate admin operations on external IP.